tests/testthat/test-merge_sce_list.R

```diff
@@ -366,12 +383,58 @@ sce_list_with_altexp <- sce_list |>
     num_altexp_features,
     total_cells / 3
   )
-
 # vector of all expected names
 full_altexp_features <- rownames(altExp(sce_list_with_altexp[[1]]))
 
 
-test_that("merging SCEs with same altExp features works as expected, with altexps", {
+
+test_that("prepare_sce_for_merge() works as expected with is_altexp=TRUE", {
+
+  test_altexp <- altExp(sce_list_with_altexp[[1]])
+  prepared_altexp <- prepare_sce_for_merge(
+    test_altexp,
+    "test",
+    batch_column = "batch",
+    cell_id_column = "cell",
+    full_altexp_features,
+    retain_coldata_cols = NULL,
+    preserve_rowdata_cols = "target_type",
+    is_altexp = TRUE
+  )
+
+  # rowdata names should start with test-, but NOT the overall column names
+  expect_true(
+    all(stringr::str_starts(colnames(rowData(prepared_altexp)), "test-"))
+  )
```

Comment:
Why not check the actual values here?
```suggestion
  expect_equal(
    colnames(rowData(prepared_altexp))
    paste0("test-", colnames(rowData(test_altexp))
  )
```

But this does not check whether we are correctly handling `preserve_rowdata_cols = "target_type"`, which we probably should... But this input doesn't have that rowdata column, so other things would have to change to check that too